quesma/quesma/search.go

```diff
@@ -213,129 +333,101 @@ func (q *QueryRunner) handleSearchCommon(ctx context.Context, indexPattern strin
+		queries := plan.Queries
+		queriesBody := make([]types.TranslatedSQLQuery, len(queries))
+		queriesBodyConcat := ""
+		for i, query := range queries {
+			queriesBody[i].Query = []byte(query.SelectCommand.String())
+			queriesBodyConcat += query.SelectCommand.String() + "\n"
 		}
-		plan.Queries, err = q.transformationPipeline.Transform(plan.Queries)
-		if err != nil {
-			logger.ErrorWithCtx(ctx).Msgf("error transforming queries: %v", err)
+		responseBody = []byte(fmt.Sprintf("Invalid Queries: %v, err: %v", queriesBody, err))
+		logger.ErrorWithCtxAndReason(ctx, "Quesma generated invalid SQL query").Msg(queriesBodyConcat)
+		bodyAsBytes, _ := body.Bytes()
+		pushSecondaryInfo(q.quesmaManagementConsole, id, "", path, bodyAsBytes, queriesBody, responseBody, startTime)
+		return responseBody, errors.New(string(responseBody))
+	}
+
+	plan.IndexPattern = indexPattern
+	plan.StartTime = startTime
+	plan.Name = model.MainExecutionPlan
+
+	var alternativePlan *model.ExecutionPlan
+
+	// TODO add alternative plan here
+
+	/* You may use this code to run alternative plan for checking how it works
+	   It breaks the tests. So, it is commented out.
+
+		alternativePlan = &model.ExecutionPlan{
+			IndexPattern:          plan.IndexPattern,
+			QueryRowsTransformers: plan.QueryRowsTransformers,
+			ResultAdapter:         plan.ResultAdapter,
+			Queries:               plan.Queries,
+			StartTime:             plan.StartTime,
+			Name:                  model.AlternativeExecutionPlan,
 		}
+	*/
 
-		plan.Queries, err = registry.QueryTransformerFor(table.Name, q.cfg, q.schemaRegistry).Transform(plan.Queries)
-		if err != nil {
-			logger.ErrorWithCtx(ctx).Msgf("error transforming queries: %v", err)
-		}
-
-		if canParse {
-			queries := plan.Queries
-			if len(queries) > 0 && query_util.IsNonAggregationQuery(queries[0]) {
-				if properties := q.findNonexistingProperties(queries[0], table, queryTranslator); len(properties) > 0 {
-					logger.DebugWithCtx(ctx).Msgf("properties %s not found in table %s", properties, table.Name)
-					if elasticsearch.IsIndexPattern(indexPattern) {
-						return queryparser.EmptySearchResponse(ctx), nil
-					} else {
-						return nil, fmt.Errorf("properties %s not found in table %s", properties, table.Name)
-					}
-				}
-			}
-			go func() {
-				defer recovery.LogAndHandlePanic(ctx, func(err error) {
-					doneCh <- AsyncSearchWithError{err: err}
-				})
-
-				translatedQueryBody, results, err := q.searchWorker(ctx, plan, table, doneCh, optAsync)
-				if err != nil {
-					doneCh <- AsyncSearchWithError{err: err}
-					return
-				}
+	var executionChan chan executionPlanResult
 
-				if len(results) == 0 {
-					logger.ErrorWithCtx(ctx).Msgf("no hits, sqls: %v", translatedQueryBody)
-					doneCh <- AsyncSearchWithError{translatedQueryBody: translatedQueryBody, err: errors.New("no hits")}
-					return
-				}
+	if alternativePlan != nil {
+		executionChan = make(chan executionPlanResult, 2)
```

Comment:
Maybe, we could have a slice with plans `plans = []string{MainExecutionPlan, AlternativeExecutionPlan} and replace const `2` with `len(plans)`

Reply:
Done